### DISCUSSION TITLE
Dashboard SQL templates

### DISCUSSION THREAD

**Discussion by ZhengchunD** [OHDSI/DataQualityDashboard#281] [Q&A] [answered]
1. Our omop database is oracle. do we need to convert the templates into oracle sql scripts?
2.  in the posted scripts, I saw { .... } in the code. is sql engine parsing the curly brace or it is R parsing it?
3. what is the logic inside the braces as shown in the following code bloc:

Your quick response and help is much appreciated as we are just starting to set up DQDashboard!
Thanks,

William Dong
.
.
plausibleGender = @plausibleGender
{@cohort}?{
cohortDefinitionId = @cohortDefinitionId
cohortDatabaseSchema = @cohortDatabaseSchema
}
**********/

SELECT num_violated_rows, CASE WHEN denominator.num_rows = 0 THEN 0 ELSE 1.0*num_violated_rows/denominator.num_rows END AS pct_violated_rows, 
  denominator.num_rows as num_denominator_rows
FROM
(
	SELECT COUNT_BIG(*) AS num_violated_rows
	FROM
	(
		SELECT cdmTable.* 
		FROM @cdmDatabaseSchema.@cdmTableName cdmTable
			INNER JOIN @cdmDatabaseSchema.person p
			ON cdmTable.person_id = p.person_id
			
			{@cohort}?{
    	JOIN @cohortDatabaseSchema.COHORT c
    	ON cdmTable.PERSON_ID = c.SUBJECT_ID
    	AND c.COHORT_DEFINITION_ID = @cohortDefinitionId
    	}
			
		WHERE cdmTable.@cdmFieldName = @conceptId
		AND p.gender_concept_id <> {@plausibleGender == 'Male'} ? {8507} : {8532} 
	) violated_rows
) violated_row_count,
( 
..

**Comment by clairblacketer** [comment, marked as answer]
@ZhengchunD thank you for posting this in discussions! So you do not need to translate into Oracle, this packages makes use of SqlRender which will translate it for you. Those curly braces are parsed by SqlRender as well, they are if/then statements that will update the SQL based on if the condition is met.

**Reply by ZhengchunD** [reply]
@clairblacketer Thank you so much for your answers! I assume those 24 templates should cover most of DQ requirements. What if we do need to add more scripts into the dashboard? Do straight mssql or oracle sql scripts work or I need to learn SqlRender first ??
Thank you.
